include/circt/Dialect/FIRRTL/FIRParser.h

```diff
@@ -78,41 +78,41 @@ void registerFromFIRFileTranslation();
 
 /// The FIRRTL specification version.
 struct FIRVersion {
-  uint32_t major, minor, patch;
-
-  /// Three way compare of one FIRRTL version with another FIRRTL version.
-  /// Return 1 if the first version is greater than the second version, -1 if
-  /// the first version is less than the second version, and 0 if the versions
-  /// are equal.
-  static int compare(const FIRVersion &a, const FIRVersion &b) {
-    if (a.major > b.major)
-      return 1;
-    if (a.major < b.major)
-      return -1;
-    if (a.minor > b.minor)
-      return 1;
-    if (a.minor < b.minor)
-      return -1;
-    if (a.patch > b.patch)
-      return 1;
-    if (a.patch < b.patch)
-      return -1;
-    return 0;
+  constexpr FIRVersion(uint16_t major, uint16_t minor, uint16_t patch)
+      : major(major), minor(minor), patch(patch) {}
```

Comment:
I was trying to think of a reason to make one of these a `uint32_t` or to consider trying to fit all of them into a `uint32_t`. However, this is not something where such optimizations make sense. (it is not duplicated many times, e.g., on every op.)